nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -55,9 +55,24 @@ int Pooling2DLayer::initialize(Manager &manager) {
     out_dim.width(in_dim.channel());
   }
 
+  if (pooling_type == PoolingType::max) {
+    max_idx.reserve(in_dim.batch() * output_dim[0].getFeatureLen());
+  } else if (pooling_type == PoolingType::global_max) {
+    max_idx_global.resize(out_dim.batch() * output_dim[0].getFeatureLen());
+  }
+
   return status;
 }
 
+void Pooling2DLayer::setBatch(unsigned int batch) {
+  Layer::setBatch(batch);
+  if (pooling_type == PoolingType::max) {
+    max_idx.reserve(batch * output_dim[0].getFeatureLen());
```

Comment:
unless you really want this to be here, please don't allocate memory in setBatch.

Reply:
redirecting duplicated discussion to https://github.com/nnstreamer/nntrainer/pull/1051#discussion_r603766020